app/Providers/FortifyServiceProvider.php

```diff
@@ -149,42 +152,56 @@ public function boot(): void
 
             $user = User::query()->where('username', $request->username)->first();
 
+            if ($user && !Hash::check($request->password, $user->password)) {
+                FailedLoginAttempt::create([
+                    'user_id'    => $user->id,
+                    'username'   => $request->username,
+                    'ip_address' => $request->ip(),
+                ]);
+
+                $user->notify(new FailedLogin(
+                    $request->ip()
+                ));
+
+                throw ValidationException::withMessages([
+                    Fortify::username() => __('auth.failed'),
+                ]);
+            }
+
             if ($user && Hash::check($request->password, $user->password)) {
                 // Check if user is activated
                 $validatingGroup = cache()->rememberForever('validating_group', fn () => Group::query()->where('slug', '=', 'validating')->pluck('id'));
 
-                if ($user->active == 0 || $user->group_id == $validatingGroup[0]) {
+                if ($user->active == 0 || $user->group_id === $validatingGroup[0]) {
                     $request->session()->invalidate();
 
                     throw ValidationException::withMessages([
-                        Fortify::username() => trans('auth.not-activated'),
+                        Fortify::username() => __('auth.not-activated'),
                     ]);
                 }
 
                 // Check if user is banned
-
```

Comment:
Personally, I add an empty line after a comment, when the comment describes code that include empty lines within it, but not going to force anything here.